### DISCUSSION TITLE
Request decompression

### DISCUSSION THREAD

**Discussion by medbak** [roadrunner-server/roadrunner#1981] [Q&A]
Hey Roadrunner community, 

I am attempting to decompress a compressed payload in flight before it reaches the application, as instructed in the official documentation.https://docs.roadrunner.dev/docs/http/gzip#documentation. 

I tried the `gzip` configuration, but it didn't work because Roadrunner is not decompressing the payload.

Is there another configuration that needs to be added, or does roadrunner not support this functionality?

Thanks,

**Comment by rustatian** [comment]
Hey @medbak 👋 
Please share your RR configuration (.rr.yaml) and a sample of your request. (cURL, Postman, etc).

**Reply by medbak** [reply]
Hey @rustatian, 

I have this RR config

```
version: '3'

server:
  command: "/usr/local/bin/php /www/applic/public/index.php"
  relay: pipes
  env:
    - APP_RUNTIME: App\Runtime\RoadRunner\Runtime

http:
  address: 0.0.0.0:${RR_HTTP_PORT:-9000}
  http2:
    h2c: true
    max_concurrent_streams: 128
  middleware: [ "gzip"]
  access_logs: true
  pool:
    num_workers: ${RR_NUM_WORKERS:-1}
    debug: false 

gzip:
    enable: true 
    level: 6 
```

The request: 

```
curl --location --request POST 'http://app.local:8000/upload/1/products' \
--header 'Content-Type: application/x-ndjson' \
--header 'Accept: application/vnd.api+json' \
--header 'Authorization: Bearer mytoken' \
--header 'Content-Encoding: gzip' \
--data-binary '@/home/sqlite/data.json.gz' 
```

The compressed file contains a json file with this data: 

```
{
    "id":"456",
    "key": "value",
    "another_key": "another_value"
}
```

Thanks

**Reply by rustatian** [reply]
Thanks, first observation is that there is no `gzip` option you tried to add to the configuration. RR packs responses starting from the 1kb IIRC (standard behavior for big bodies), but do not touch request bodies.
You may create a feature request (roadrunner repo) or send us a pull request to the github.com/roadrunner-server/gzip plugin. If you go with PR I think (sorry for errors, writing here in text 😄 ) you may write something similar to:
```go
        if strings.Contains(r.Header.Get("Content-Encoding"), "gzip") {
            gz, err := gzip.NewReader(r.Body)
            if err != nil {
                // on error write log (warning) and send ungzipped request to the next middleware
            }
            defer gz.Close()
            r.Body = io.NopCloser(gz) // <-- you'll need to replace the actual body, since body should be sent to the worker.
        }
        ```

**Reply by medbak** [reply]
Thanks @rustatian for your reply
